### ISSUE TITLE
Default CodeMirror themes merged with application styles

### ISSUE THREAD

**Issue by matthanley** [sparksuite/simplemde-markdown-editor#318]
Default CodeMirror themes are overridden by the CSS distributed with SMDE. A more flexible approach would be to implement a separate theme for CodeMirror in the distributed CSS and set this theme on the CodeMirror instance by default.

**Comment by SvenAlHamad** [comment]
I agree ... I also have the issue where SMDE theme overwrites my CodeMirror theme:

Before SMDE is added:
![screen shot 2016-05-19 at 20 25 45](https://cloud.githubusercontent.com/assets/3808420/15406822/ee6737c2-1dff-11e6-9c54-ded777f43d81.png)

After SMDE is added:
![screen shot 2016-05-19 at 20 28 03](https://cloud.githubusercontent.com/assets/3808420/15406890/3e5b9818-1e00-11e6-991c-682300e3fc2c.png)

Basically just by wrapping the SMDE into its own element with a common class, e.g. `.smde` you can then change the selectors in the SMDE css from this `.CodeMirror .CodeMirror-code .cm-header-3` into `.smde .CodeMirror .CodeMirror-code .cm-header-3` which will sort out any conflicts amongst themes.

UPDATE:
Just tested my theory and can confirm it's working as expected. Here is the end result, CodeMirror and SMDE side-by-side on the same page:
![screen shot 2016-05-19 at 20 41 37](https://cloud.githubusercontent.com/assets/3808420/15407331/36e6d078-1e02-11e6-8dbb-7bd1be792a63.png)

UPDATE no.2:
Seems there are still some conflicts, but on the code level ... SMDE somehow overwrites the CodeMirror configuration in a way that it doesn't include all the modes. You can see from my previous screen that the theme looks good, but the highlighting on css code doesn't work, while when I remove SMDE, the highlight is working properly.

**Comment by WesCossick** [comment]
Probably something that would be good for the future version 2 redesign.
